src/lib/storage/constraints/foreign_key_constraint.cpp

```diff
@@ -0,0 +1,121 @@
+#include "foreign_key_constraint.hpp"
+
+#include <numeric>
+
+#include <boost/container_hash/hash.hpp>
+
+namespace {
+
+using namespace hyrise;  // NOLINT(build/namespaces)
+
+// ForeignKeyConstraints with swapped columns are equivalent. To ensure they are represented the same way, we determine
+// the permutation that sorts `foreign_key_columns` and apply it to `foreign_key_columns` and `primary_key_columns`.
```

Comment:
To transform `[b, a] in [y ,x]` to `[a, b] in [x, y]` and not to `[b, a] in [x, y]` (which would now mean every value of `b` is also in `x` instead of `y`).